### PR TITLE
added: eps output functions for LR nurbs

### PR DESCRIPTION
**Pull request by akva2**
We need to duplicate this code from libLR since they evaluate the spline basis.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/LR/ASMu2Dnurbs.C [hunk 1]
```diff
@@ -289,3 +289,183 @@ void ASMu2D::computeBasisNurbs (double u, double v,
...
+  // outputting eps files.
+  this->generateBezierBasis();
+  this->generateBezierExtraction();
+
+  // get bounding box (max/min of the control points)
+  std::array<double,2> x{1e7, -1e7};
+  std::array<double,2> y{1e7, -1e7};
+  for (const LR::Basisfunction* b : lrspline->getAllBasisfunctions()) {
+      std::vector<double>::const_iterator cp = b->cp();
+      x[0] = (cp[0] < x[0]) ? cp[0] : x[0];
```
Comment (kmokstad) [inline src/ASM/LR/ASMu2Dnurbs.C:320]: operator[] on an const_iterator is well-defined, I assume?

**Review by kmokstad — APPROVED** [review]
Looks fine, except for some cosmetics fixes here and there, and a suggestion.

File: src/ASM/LR/ASMu2Dnurbs.C [hunk 2]
```diff
@@ -289,3 +289,183 @@ void ASMu2D::computeBasisNurbs (double u, double v,
...
+{
+  // get date
+  time_t t = time(0);
+  tm* lt = localtime(&t);
+  char date[128];
+  sprintf(date, "%02d/%02d/%04d", lt->tm_mday, lt->tm_mon + 1, lt->tm_year+1900);
+
+  if (mesh != lrspline) {
+    mesh.swap(lrspline);
+    geo = lrspline.get();
```
[the remark this answers is not in the document]
Comment (akva2) [reply src/ASM/LR/ASMu2Dnurbs.C:305]: there's unfortunately more to it than the bezier extraction. it also uses the geo pointer here and there (in the ASMu2D class). at some point that can be cleared up but it's not straight forward.

File: src/ASM/LR/ASMu2Dnurbs.C [hunk 3]
```diff
@@ -289,3 +289,183 @@ void ASMu2D::computeBasisNurbs (double u, double v,
...
+
+  // get bounding box (max/min of the control points)
+  std::array<double,2> x{1e7, -1e7};
+  std::array<double,2> y{1e7, -1e7};
+  for (const LR::Basisfunction* b : mesh->getAllBasisfunctions()) {
+      std::vector<double>::const_iterator cp = b->cp();
+      x[0] = (cp[0] < x[0]) ? cp[0] : x[0];
+      x[1] = (cp[0] > x[1]) ? cp[0] : x[1];
+      y[0] = (cp[1] < y[0]) ? cp[1] : y[0];
+      y[1] = (cp[1] > y[1]) ? cp[1] : y[1];
```
[the remark this answers is not in the document]
Comment (akva2) [reply src/ASM/LR/ASMu2Dnurbs.C]: fixed.

File: src/ASM/LR/ASMu2Dnurbs.C [hunk 4]
```diff
@@ -289,3 +289,183 @@ void ASMu2D::computeBasisNurbs (double u, double v,
...
+    int textX   = ((*b)[0][1] == (*b)[0][mesh->order(0)]) ? -2 : 1;
+    int textY   = ((*b)[1][1] == (*b)[1][mesh->order(1)]) ? -2 : 1;
+    // move text on edge functions
+    if ((*b)[0][1] == mesh->endparam(0))
+      textX = 1;
+    else if ((*b)[0][mesh->order(0)-1] == mesh->startparam(0))
+      textX = -2;
+    if ((*b)[1][1] == mesh->endparam(1))
+      textY = 1;
+    else if((*b)[1][mesh->order(1)-1] == mesh->startparam(1))
```
[the remark this answers is not in the document]
Comment (akva2) [reply src/ASM/LR/ASMu2Dnurbs.C]: darn, forgot one (no spaces in code i copied from..)

File: src/ASM/LR/ASMu2Dnurbs.C (hunk 1 above)
Comment (akva2) [reply src/ASM/LR/ASMu2Dnurbs.C]: yes it is.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/LR/ASMu2Dnurbs.C (hunk 2 above)
[the remark this answers is not in the document]
Comment (kmokstad) [reply src/ASM/LR/ASMu2Dnurbs.C:301]: ok, the lrspline pointer is used in the evalPoint method, which is invoked here. Then leave it like this for now.

**Pull request merged**